src/llm/apis/openai_completions.cpp

```diff
@@ -259,8 +261,6 @@ absl::Status OpenAIChatCompletionsHandler::parseMessages() {
                     if (member->value.IsString()) {
                         request.chatHistory.back().insert({member->name.GetString(), member->value.GetString()});
                     }
-                } else {
-                    return absl::InvalidArgumentError("Invalid message structure - content should be string or array");
```

Comment:
Why is this removed?

Reply:
before, there was assumption that only the fields can be in the message, role and content. it is not the case with tool role.

Reply:
This path is executed when content is not an array and it looks to me that it's relevant part of the logic we had so far.
I would make explicit exception for `tool` role instead